tests/ui/associated-type-bounds/return-type-notation/not-a-method.stderr

```diff
@@ -0,0 +1,49 @@
+error[E0575]: expected function, found function `function`
```

Comment:
Namely, this is a consequence of E0575 calling associated functions "functions". 

As a follow-up, I will do like https://github.com/rust-lang/rust/pull/130414 and probably split these resolver errors into their own error codes, so they can have tailored explanations.